src/debug_utils.cc

```diff
@@ -299,6 +320,103 @@ void CheckedUvLoopClose(uv_loop_t* loop) {
   CHECK(0 && "uv_loop_close() while having open handles");
 }
 
+std::vector<std::string> NativeSymbolDebuggingContext::GetLoadedLibraries() {
+  std::vector<std::string> list;
+#ifdef __linux__
+  dl_iterate_phdr(
+      [](struct dl_phdr_info* info, size_t size, void* data) {
+        auto list = static_cast<std::vector<std::string>*>(data);
+        if (*info->dlpi_name != '\0') {
+          list->push_back(info->dlpi_name);
+        }
+        return 0;
+      },
+      &list);
+#elif __APPLE__
+  uint32_t i = 0;
+  for (const char* name = _dyld_get_image_name(i); name != nullptr;
+       name = _dyld_get_image_name(++i)) {
+    list.push_back(name);
+  }
+
+#elif _AIX
+  // We can't tell in advance how large the buffer needs to be.
+  // Retry until we reach too large a size (1Mb).
+  const unsigned int kBufferGrowStep = 4096;
+  MallocedBuffer<char> buffer(kBufferGrowStep);
+  int rc = -1;
+  do {
+    rc = loadquery(L_GETINFO, buffer.data, buffer.size);
+    if (rc == 0) break;
+    buffer = MallocedBuffer<char>(buffer.size + kBufferGrowStep);
+  } while (buffer.size < 1024 * 1024);
+
+  if (rc == 0) {
+    char* buf = buffer.data;
+    ld_info* cur_info = nullptr;
+    do {
+      std::ostringstream str;
+      cur_info = reinterpret_cast<ld_info*>(buf);
+      char* member_name = cur_info->ldinfo_filename +
+          strlen(cur_info->ldinfo_filename) + 1;
+      if (*member_name != '\0') {
+        str << cur_info->ldinfo_filename << "(" << member_name << ")";
+        list.push_back(str.str());
+        str.str("");
+      } else {
+        list.push_back(cur_info->ldinfo_filename);
+      }
+      buf += cur_info->ldinfo_next;
+    } while (cur_info->ldinfo_next != 0);
+  }
+#elif __sun
+  Link_map* p;
+
+  if (dlinfo(RTLD_SELF, RTLD_DI_LINKMAP, &p) != -1) {
+    for (Link_map* l = p; l != nullptr; l = l->l_next) {
+      list.push_back(l->l_name);
+    }
+  }
+
+#elif _WIN32
+  // Windows implementation - get a handle to the process.
+  HANDLE process_handle = OpenProcess(PROCESS_QUERY_INFORMATION|PROCESS_VM_READ,
+                                      FALSE, GetCurrentProcessId());
+  if (process_handle == nullptr) {
+    // Cannot proceed, return an empty list.
+    return list;
+  }
+  // Get a list of all the modules in this process
+  DWORD size_1 = 0;
+  DWORD size_2 = 0;
+  // First call to get the size of module array needed
+  if (EnumProcessModules(process_handle, nullptr, 0, &size_1)) {
+    MallocedBuffer<HMODULE> modules(size_1);
+
+    // Second call to populate the module array
+    if (EnumProcessModules(process_handle, modules.data, size_1, &size_2)) {
+      for (DWORD i = 0;
+           i < (size_1 / sizeof(HMODULE)) && i < (size_2 / sizeof(HMODULE));
+           i++) {
+        WCHAR module_name[MAX_PATH];
+        // Obtain and report the full pathname for each module
+        if (GetModuleFileNameEx(process_handle,
```

Comment:
Maybe `GetModuleFileNameExW ` to fix the compilation error?

Reply:
thanks @bzoz . I have 2 options;
 - use `TCHAR` instead of wide characters: I am able to build it successfully with that. 
 - use `GetModuleFileNameExW` and deal with the next error of converting the result into `std::string` through `MultiByteToWideChar`.

On an assumption that the module names wont be made with wide characters, do you think the second option is an overkill, and we can just go with TCHAR instead?  pls let me know.

Reply:
> an assumption that the module names wont be made with wide characters

I’d definitely prefer not to make such an assumption…